### ISSUE TITLE
Polling vs non polling

### ISSUE THREAD

**Issue by vchuravy** [JuliaParallel/UCX.jl#36]
Currently UCX.jl uses [`FileWatching.poll_fd`](https://github.com/JuliaParallel/UCX.jl/blob/2bac1cc839430997503099f7514f682ef90d4432/src/UCX.jl#L264-L312) in order to wait on incoming messages, this relies on Julia usage of libuv to allow the watcher task to suspend. This is the preferred approach since it doesn't introduce a Julia task that looks artificially busy to the scheduler.

This does depend on [`UCP_FEATURE_WAKEUP`](https://github.com/JuliaParallel/UCX.jl/blob/2bac1cc839430997503099f7514f682ef90d4432/src/UCX.jl#L165) which currently leads to shared memory not being used by UCX (https://github.com/openucx/ucx/issues/5322)

In symmetric use-cases like tagged messages and streams not using polling is fine, but in asymmetric use-cases like active messages polling is essential, otherwise we will not call `progress` on the `UCXWorker` often enough. Even with polling I have seen timing improvements by switching the watcher task into busy waiting (which is bad for the rest of the system).

Alternatives that I thought about:

- LibUV timer, but wake ups occur not often enough
- LibUV `idle`https://github.com/JuliaParallel/UCX.jl/commit/276874d7905157b4ef7ca0d37a1cf29cbb243f7a -- need to benchmark again.
- `@threadcall`: `progress` will invoke callbacks into Julia code, can't execute that on a libuv thread

**Comment by vchuravy** [comment]
Some numbers with vc/polling_test

| Configuration | Median | Mean | STD |
| -----------------  | ---------- | ------- | ------ |
| Polling            | 23.455 μs | 25.666 μs |  6.095 μs | 
| Busy              | 4.260 μs |  4.654 μs | 1.081 μs | 

This is still using `UCP_FEATURE_WAKEUP` and I am not using CMA on my local x86 machine.

**Comment by vchuravy** [comment]
X86:
```
julia> @benchmark wait(@async(nothing))
BenchmarkTools.Trial: 
  memory estimate:  704 bytes
  allocs estimate:  4
  --------------
  minimum time:     812.603 ns (0.00% GC)
  median time:      857.945 ns (0.00% GC)
  mean time:        900.042 ns (2.29% GC)
  maximum time:     21.296 μs (94.88% GC)
  --------------
  samples:          10000
  evals/sample:     73

```

PPC:
```
julia> @benchmark wait(@async(nothing))
BenchmarkTools.Trial: 
  memory estimate:  1.22 KiB
  allocs estimate:  4
  --------------
  minimum time:     2.911 μs (0.00% GC)
  median time:      3.090 μs (0.00% GC)
  mean time:        3.121 μs (0.74% GC)
  maximum time:     144.288 μs (97.58% GC)
  --------------
  samples:          10000
  evals/sample:     9
```

**Comment by vchuravy** [comment]
fixed in #39
